### PR TITLE
Add temporary fix for ray import pandas race condition

### LINKED ISSUE #1175: Race condition on imports with Ray
On Ray 0.8.3, there is a race condition that will break the imports. Relevant thread in the Ray repo: ray-project/ray#7879

### PR DESCRIPTION
**Pull request by simon-mo**
## What do these changes do?
Attempt to address issues raised in https://github.com/ray-project/ray/issues/7879

- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1175 
- [x] tests ~added and~ passing

### CODE REVIEW

**Comment by robertnishihara** [comment]
@simon-mo @devin-petersohn does this appear tofix the issue?

**Comment by gshimansky** [comment]
> @simon-mo @devin-petersohn does this appear to work around the issue?

I tried running tests on Ray several times and didn't have any fails even with 100 pytest workers. So I think the issue is indeed fixed. TeamCity confit is probably wrong and that's the reason why CI testing failed. I restarted it with modified config.

**Comment by simon-mo** [comment]
looks like it's passing the ray test? is there anyway to keep re-run the teamcity build multiple time?

**Comment by gshimansky** [comment]
> looks like it's passing the ray test? is there anyway to keep re-run the teamcity build multiple time?

Yes Ray test passed with 48 workers (TeamCity system has 48 CPU cores). I don't know a way to repeat TC tests other than to restart them manually, but as I wrote before, I ran tests on another system with 100 workers and they all passed to, so I think this race condition is really fixed with this change.

File: modin/pandas/__init__.py [hunk 1]
```diff
@@ -226,6 +226,14 @@ def move_stdlib_ahead_of_site_packages(*args):
             move_stdlib_ahead_of_site_packages
         )
 
+        # Register a fix to import pandas on all workers before running tasks.
+        # This prevents a race condition between two threads deserializing functions
+        # and trying to import pandas at the same time.
+        def import_pandas(*args):
+            import pandas
```
Comment (gshimansky) [inline modin/pandas/__init__.py:233]: Can you please change the line like this to tell flake8 lint to ignore the error?
```python
    import pandas  # noqa F401
````
two spaces between code and `#` symbol are required for this to work.

**Pull request merged**